python/sglang/srt/server.py

```diff
@@ -123,31 +123,91 @@ async def flush_cache():
     )
 
 
-async def detokenize_logprob_tokens(token_logprobs):
-    token_ids = [tid for tid, _ in token_logprobs]
+async def detokenize_logprob_tokens(token_logprobs, decode_to_text):
+    if not decode_to_text:
+        return [(logprob, token_id, None) for logprob, token_id in token_logprobs]
+
+    token_ids = [tid for _, tid in token_logprobs]
     token_texts = await tokenizer_manager.detokenize(DetokenizeReqInput(token_ids))
-    return [(text, logprob) for text, (_, logprob) in zip(token_texts, token_logprobs)]
+    return [
+        (logprob, token_id, token_text)
+        for (logprob, token_id), token_text, in zip(token_logprobs, token_texts)
+    ]
+
+
+async def detokenize_top_logprobs_tokens(top_logprobs, decode_to_text):
+    for i, t in enumerate(top_logprobs):
+        if top_logprobs[i] is not None:
+            top_logprobs[i] = await detokenize_logprob_tokens(t, decode_to_text)
+    return top_logprobs
+
+
+async def handle_token_logprobs_results(obj: GenerateReqInput, ret):
+    """
+    Convert the style `(logprobs, token_id)` into `(logprobs, token_id, token_text)`
+    """
+    # NOTE: This is because the multiple requests in one http request.
```

Comment:
This comment can be put into the function docstring directly and refined to something like `This function supports multiple internal requests created by a single HTTP request`.